### PR TITLE
Update the tasks endpoints

### PR DESCRIPTION
**Pull request by Kerollmops**
This PR wraps all the changes related to the `tasks` endpoints, it is related to https://github.com/meilisearch/meilisearch/issues/2377 but doesn't close it. I will create a new PR to work on [the seek-based pagination](https://github.com/meilisearch/specifications/pull/115).

I wanted to do something cool with Github: being able to merge multiple PR in this one, to help review changes one by one, unfortunately, Github doesn't allow creating empty PRs. I also struggled with git itself when it comes to merging things in the right order, so I decided that I would add all of the changes in this single PR. I will list the changes and references to the specs here.

 - [x] Tasks statuses and types must be case insensitive
 - [x] Tasks statuses, types and indexUid must accept the `*` selector
 - [ ] Rename the `TaskDetails` struct fields

## Changes

- [ ] ~Add seek-based pagination following [the spec](https://github.com/meilisearch/specifications/pull/115)~ Moved in #2445. 
- [x] Add filtering on the `/tasks` endpoint following [this spec](https://github.com/meilisearch/specifications/pull/116)
  - [x] Add filtering capabilities on `type`, `status` and `indexUid` for `GET` `task` lists endpoints.
  - [x] It is possible to specify several values for a filter using the `,` character. e.g. `?status=enqueued,processing`
  - [x] Between two different filters, an AND operation is applied. e.g. `?status=enqueued&type=indexCreation` is equivalent to `status=enqueued AND type = indexCreation`
- [x] Remove `GET /indexes/:indexUid/tasks`. It can be replaced by `GET /tasks?indexUid=:indexUid`
- [x] Remove `GET /indexes/:indexUid/tasks/:taskUid`.
- [x] Rename `uid` to `taskUid` in the `202 - Accepted` task response return by every asynchronous tasks (ex: index creation, document addition...)
- [x] Rename some task properties
  - [x] `documentPartial`-> `documentAdditionOrUpdate`
  - [x] `documentAddition`-> `documentAdditionOrUpdate`
  - [x] `clearAll` -> `documentDeletion`

### CODE REVIEW

File: meilisearch-http/src/task.rs [hunk 1]
```diff
@@ -79,15 +84,19 @@ pub enum TaskStatus {
 }
 
 impl FromStr for TaskStatus {
-    type Err = &'static str;
+    type Err = String;
 
-    fn from_str(status: &str) -> Result<Self, &'static str> {
+    fn from_str(status: &str) -> Result<Self, String> {
```
Comment (MarinPostma) [inline meilisearch-http/src/task.rs:83]: Same here, I think we need to return a custom error.

**Review by MarinPostma — CHANGES_REQUESTED** [review]

File: meilisearch-http/src/routes/tasks.rs [hunk 2]
```diff
@@ -68,6 +107,42 @@ async fn get_tasks(
...
+            let mut filters = indexes_filters.unwrap_or_default();
+            filters.filter_fn(move |task| {
+                statuses
+                    .iter()
+                    .any(|s| task_status_matches_events(&s, &task.events))
+            });
+            Some(filters)
+        }
+        (None, None) => indexes_filters,
+    };
```
[the remark this answers is not in the document]
Comment (MarinPostma) [reply meilisearch-http/src/routes/tasks.rs]: This is a lot of repeated code though, another alternative is to move the match inside the closure


```rust
if type_.is_some() || status.is_some() {
    filter.filter_fn(move || {
        let pass_type = match type_ {
            ...
       };
        let pass_status  = match status {
            ...
       };
       pass_status && pass_type
})
```

[the remark this answers is not in the document]
Comment (Kerollmops) [reply meilisearch-http/src/routes/tasks.rs]: That's a good and better idea, you are right, I'll try this. Thank you!

File: meilisearch-http/src/task.rs [hunk 3]
```diff
@@ -1,64 +1,104 @@
...
+        } else {
+            Err(format!(
+                "invalid task type `{}`, expecting one of: \
+                indexCreation, indexUpdate, indexDeletion, documentAdditionOrUpdate, \
+                documentDeletion, settingsUpdate, dumpCreation",
+                status
+            ))
+        }
+    }
+}
```
Comment (MarinPostma) [inline meilisearch-http/src/task.rs:69]: ping @gmourier I'm thinking that since we are case insensitive on that, maybe we should also be on ranking rules?

**Review by MarinPostma — APPROVED** [review]
👌 

bors merge

**Pull request merged**